lib/SimpleSAML/Kernel.php

```diff
@@ -0,0 +1,175 @@
+
+    const CONFIG_EXTS = '.{php,xml,yaml,yml}';
+
+    /**
+     * @var string
+     */
+    private $module;
+
+
+    /**
+     * @oaram string $module
+     */
+    public function __construct($module)
+    {
+        $this->module = $module;
+
+        parent::__construct('dev', false);
+    }
+
+
+    /**
+     * @return string
+     */
+    public function getCacheDir()
+    {
+        $configuration = Configuration::getInstance();
+        $cachePath = $configuration->getString('tempdir').'/cache/'.$this->module;
+
+        if (0 === strpos($cachePath, '/')) {
+            return $cachePath;
+        }
+
+        return $configuration->getBaseDir().'/'. $cachePath;
+    }
+
+
+    /**
+     * @return string
+     */
+    public function getLogDir()
+    {
+        $configuration = Configuration::getInstance();
+        $loggingPath = $configuration->getString('loggingdir');
+
+        if (0 === strpos($loggingPath, '/')) {
+            return $loggingPath;
+        }
+
+        return $configuration->getBaseDir().'/'. $loggingPath;
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    public function registerBundles()
+    {
+        return [
+            new FrameworkBundle(),
+        ];
+    }
+
+
+    /**
+     * Get the module loaded in this kernel.
+     *
+     * @return string
+     */
+    public function getModule()
+    {
+        return $this->module;
+    }
+
+
+    /**
+     * Configures the container.
+     *
+     * @param ContainerBuilder $container
+     * @param LoaderInterface $loader
+     * @return void
+     */
+    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader)
+    {
+        $confDir = __DIR__;
+        $loader->load($confDir . '/Resources/config/services/*' . self::CONFIG_EXTS, 'glob');
+        $confDir = Module::getModuleDir($this->module) . '/config/services';
+        if (is_dir($confDir)) {
+            $loader->load($confDir . '/**/*' . self::CONFIG_EXTS, 'glob');
+        }
+
+        $container->loadFromExtension('framework', [
+            'secret' => Configuration::getInstance()->getString('secretsalt'),
+        ]);
+
+        $this->registerModuleControllers($container);
+    }
+
+
+    /**
+     * Import routes.
+     *
+     * @param RouteCollectionBuilder $routes
+     * @return void
+     */
+    protected function configureRoutes(RouteCollectionBuilder $routes)
+    {
+        $confDir = __DIR__;
+        $routes->import($confDir . '/Resources/config/routes/*' . self::CONFIG_EXTS, '/', 'glob');
+        $confDir = Module::getModuleDir($this->module) . '/config/routes';
+        if (is_dir($confDir)) {
+            $routes->import($confDir . '/**/*' . self::CONFIG_EXTS, $this->module, 'glob');
+        }
+    }
+
+
+    /**
+     * @param $container
+     * @return void
+     */
+    private function registerModuleControllers($container)
+    {
+        try {
+            $definition = new Definition();
+            $definition->setAutowired(true);
+
```

Comment:
In my branch I was using symfony 3 and I guess you are using symfony 4. In sf4 all services are private by default. But here the message is pretty confusing, because I don't understand why say you forgot to add a tag.

Anyway try to tag the definition:

```
$definition->addTag('controller.service_arguments');
```

or do it public

```
$definition->setPublic(true);
```

Reply:
I tried both options, but it doesn't work because `registerModuleControllers` is never called..
Could it be that you never finished this?

Reply:
For some reason, if I set debug=true it starts working... Setting it to false breaks it again..

Reply:
It's something in the parent-constructor of the Kernel-class that triggers this..  And I can't pinpoint it..

Reply:
This Symfony-code is pretty hard to follow.. I can pinpoint the crappy flow up until https://github.com/symfony/http-kernel/blob/9979a7022d80fb0c9a6dcf30830e073a1bc101c7/Controller/ControllerResolver.php#L131 but there's something hidden behind that.. I don't see why it would fail, the classname is valid.. The error isn't triggered from this line, so it must somehow be hooked to other code.. Also, if I set debug=true, this particular code is never reached at all, making this very hard to pinpoint

Reply:
It seems to have been an unfortunate case of coincidence..
Restarting httpd & php-fpm magically solved my issues...

Reply:
Good to know. Sorry, I'm on vacation and I can not follow the issue. I have the problem than I haven't a development environment to test this (I tried but it requires kerberos for some module).

A docker-compose/Dockerfile  with everything ready to test could be useful to help.

Reply:
Hi @sgomez!  Enjoy your vacation!
If you can't instal the php-krb5 module on your development environment, you can now remove the negotiate-module from the composer.json file..
I will try and run some more tests, but I'm confident that the issue is solved and something was messed up in cache. Unfortunately I'm too unfamiliar with Docker to help you with that.

Reply:
Just a quick note regarding docker; I'm in the midst of preparing just that. It will be pushed here; https://github.com/simplesamlphp/docker-simplesamlphp hopefully by the end of today.